binder.go

```diff
@@ -0,0 +1,1212 @@
+func (b *ValueBinder) Durations(sourceParam string, dest *[]time.Duration) *ValueBinder {
+	return b.durationsValue(sourceParam, dest, false)
+}
+
+// MustDurations requires parameter values to exist to be bind to slice of time.Duration variables. Returns error when values does not exist
+func (b *ValueBinder) MustDurations(sourceParam string, dest *[]time.Duration) *ValueBinder {
+	return b.durationsValue(sourceParam, dest, true)
+}
+
+func (b *ValueBinder) durationsValue(sourceParam string, dest *[]time.Duration, valueMustExist bool) *ValueBinder {
+	if b.failFast && b.errors != nil {
+		return b
+	}
+
+	values := b.ValuesFunc(sourceParam)
+	if len(values) == 0 {
+		if valueMustExist {
+			b.setError(b.ErrorFunc(sourceParam, []string{}, "required field value is empty", nil))
+		}
+		return b
+	}
+	return b.durations(sourceParam, values, dest)
+}
+
+func (b *ValueBinder) durations(sourceParam string, values []string, dest *[]time.Duration) *ValueBinder {
+	tmp := make([]time.Duration, len(values))
+	for i, v := range values {
+		t, err := time.ParseDuration(v)
+		if err != nil {
+			b.setError(b.ErrorFunc(sourceParam, []string{v}, "failed to bind field value to Duration", err))
+			if b.failFast {
+				return b
+			}
+			continue
+		}
+		tmp[i] = t
+	}
+	if b.errors == nil {
+		*dest = tmp
+	}
+	return b
+}
+
+// UnixTime binds parameter to time.Time variable (in local Time corresponding to the given Unix time).
+// NB: value larger than 2147483647 is interpreted as UnixNano (unix time with nanosecond precision) where
+// value last 9 characters (e.g. `999999999`) are nano seconds and everything before them are seconds part.
+func (b *ValueBinder) UnixTime(sourceParam string, dest *time.Time) *ValueBinder {
+	return b.unixTime(sourceParam, dest, false)
+}
+
+// MustUnixTime requires parameter value to exist to be bind to time.Duration variable  (in local Time corresponding
+// to the given Unix time). Returns error when value does not exist.
+// NB: value larger than 2147483647 is interpreted as UnixNano (unix time with nanosecond precision) where
+// value last 9 characters (e.g. `999999999`) are nano seconds and everything before them are seconds part.
+func (b *ValueBinder) MustUnixTime(sourceParam string, dest *time.Time) *ValueBinder {
+	return b.unixTime(sourceParam, dest, true)
+}
+
+func (b *ValueBinder) unixTime(sourceParam string, dest *time.Time, valueMustExist bool) *ValueBinder {
+	if b.failFast && b.errors != nil {
+		return b
+	}
+
+	value := b.ValueFunc(sourceParam)
+	if value == "" {
+		if valueMustExist {
+			b.setError(b.ErrorFunc(sourceParam, []string{value}, "required field value is empty", nil))
+		}
+		return b
+	}
+
+	n, err := strconv.ParseInt(value, 10, 64)
+	if err != nil {
+		b.setError(b.ErrorFunc(sourceParam, []string{value}, "failed to bind field value to Time", err))
+		return b
+	}
+
+	if n <= 2147483647 {
+		*dest = time.Unix(n, 0)
+	} else {
+		*dest = time.Unix(0, n)
```

Comment:
For **999999999** this will be **1970-01-01 00:00:00.999999999 +0000 UTC**
Could you please provide a use case for this than can help me to understand it?

Reply:
This is to allow nano second precision timestamps. I do my golang work mostly in *IoT* related field and from client to client requirements are different - for some clients data (ala GPS location) should be stored to timeserie database at second precision (for that we pass around unixtimestamp) and sometimes (ala accelerometers/gyros)  at more precise granularity (for that unixnano is suitable). Passing integers (unixtimestamps) around instead of date strings is more convenient, especially unixtimestamp as it is UTC and you do not have to think about timezones. As `time.Unix(s,n)` supports both and when building data fetching API you would (I would) want to use both (conviently) on same field. `2147483647` is deciding point as it is max unixtimestamp per int32 value. uint32 max `4294967295` value could do also but bigger values would start to overlap with `999999999`

It is mentioned in method docs
```
// UnixTime binds parameter to time.Time variable (in local Time corresponding to the given Unix time).
// NB: value larger than 2147483647 is interpreted as UnixNano (unix time with nanosecond precision) where
// value last 9 characters (e.g. `999999999`) are nano seconds and everything before them are seconds part.
func (b *ValueBinder) UnixTime(sourceParam string, dest *time.Time) *ValueBinder {
```

to support standard library nano argument https://github.com/golang/go/blob/9eef49cfa6eb016e3b20df189e540c6c5a71f365/src/time/time.go#L1292

Reply:
but this is just me. maybe for clarity sake sec and nano should be different methods

Reply:
Thanks for the use case, it help me a lot to understand it better